src/frontends/onnx/frontend/src/op/layer_normalization.cpp

```diff
@@ -43,58 +32,42 @@ ov::OutputVector layer_normalization(const ov::frontend::onnx::Node& node) {
                      num_inputs);
+    CHECK_VALID_NODE(node,
+                     node.get_outputs_size() == 1,
+                     "LayerNormalization expects 1 output tensor to be used in a model, other configurations are used "
+                     "for training and are not supported. Got: ",
+                     node.get_outputs_size(),
+                     " outputs.");
 
-    const auto& X = inputs.at(0);
-    const auto& Scale = inputs.at(1);
-
-    auto axis = node.get_attribute_value<std::int64_t>("axis", -1);
-    double epsilon = node.get_attribute_value<double>("epsilon", 1e-5);
-    int64_t stash_type_i =
-        node.get_attribute_value<int64_t>("stash_type",
-                                          static_cast<int64_t>(TensorProto_DataType::TensorProto_DataType_FLOAT));
+    auto default_stash_type_i = static_cast<int64_t>(TensorProto_DataType::TensorProto_DataType_FLOAT);
+    int64_t stash_type_i = node.get_attribute_value<int64_t>("stash_type", default_stash_type_i);
     element::Type stash_type = common::get_ov_element_type(stash_type_i);
+    CHECK_VALID_NODE(node,
+                     stash_type == element::f32 || stash_type == element::bf16,
+                     "LayerNormalization `stash_type` attribute is expected to represent float or bfloat16 type. Got: ",
+                     stash_type);
 
-    // following calculations are kept as close to the onnx\defs.cc description as possible
-    auto FloatEpsilon = Constant::create(ov::element::f32, Shape{}, {epsilon});
-    auto Epsilon = std::make_shared<Convert>(FloatEpsilon, stash_type);
-    auto XShape = std::make_shared<ShapeOf>(X);
-    auto Rank = std::make_shared<v3::ShapeOf>(XShape);
-    auto Zero1D = Constant::create(ov::element::i64, {1}, {0});
-    auto One1D = Constant::create(ov::element::i64, {1}, {1});
-    auto Axis1D = Constant::create(ov::element::i64, {1}, {axis});
-    auto PrefixShape = std::make_shared<Slice>(XShape, Zero1D, Axis1D, One1D);
-    ov::Output<ov::Node> NumReducedAxes = (axis >= 0 ? std::make_shared<Subtract>(Rank, Axis1D)->output(0)
-                                                     : std::make_shared<Negative>(Axis1D)->output(0));
-    auto SuffixShape = std::make_shared<v3::Broadcast>(One1D, NumReducedAxes);
-    auto ReducedShape = std::make_shared<Concat>(ov::OutputVector{PrefixShape, SuffixShape}, 0);
-
-    auto X2D = ov::op::util::flatten(X, static_cast<int>(axis));
-    auto XU = std::make_shared<Convert>(X2D, stash_type);
-
-    auto Mean2D = std::make_shared<ReduceMean>(XU, One1D, true);
-    auto Square = std::make_shared<Multiply>(XU, XU);
-    auto MeanOfSquare = std::make_shared<ReduceMean>(Square, One1D, true);
-    auto SquareOfMean = std::make_shared<Multiply>(Mean2D, Mean2D);
+    ov::Output<ov::Node> data = inputs.at(0);
+    element::Type original_type = data.get_element_type();
+    bool needs_type_casting = original_type.is_dynamic() || stash_type != original_type;
```

Comment:
Can we just compare with dynamic type without checking that it is dynamic?

Reply:
Updated code. Please review